### PR TITLE
Fix test failure related to numpy 2.4 warning

### PR DESCRIPTION
**Pull request by Nanored4498**
As mentioned in https://github.com/joblib/joblib/pull/1768#issuecomment-3723153945 there is a recent numpy version 2.4.0 added a new warning when `align=` is not passed as boolean to `np.dtype()`.
This warning should be handled in the pickle test across python versions.

### CODE REVIEW

File: joblib/test/test_numpy_pickle.py [hunk 1]
```diff
@@ -416,6 +416,22 @@ def _check_pickle(filename, expected_list, mmap_mode=None):
                 if (re.search("_0.1.+.pkl$", filename_base) and mmap_mode is not None)
                 else 0
             )
+
+            if np.__version__.split(".") >= "2.4.0".split("."):
```
Comment (tomMoral) [inline joblib/test/test_numpy_pickle.py]: You can simply compare the string, lexicographic order will be the same
```suggestion
            if np.__version__ >= "2.4.0":
```

**Review by tomMoral — APPROVED** [review]

File: joblib/test/test_numpy_pickle.py (hunk 1 above)
Comment (Nanored4498) [reply joblib/test/test_numpy_pickle.py]: It depends if a version 2.10 is released. If so we will have "2.10" < "2.4".

Comment (tomMoral) [reply joblib/test/test_numpy_pickle.py]: But it might now be always int (for instance with `rc`; Agreed it is a corner case...)

**Review by tomMoral — APPROVED** [review]
Just a nitpick (don't add a function that is not used multiple time) but otherwise LGTM.

I did some clarification on variable names/check order directly as I could not put suggestion (don't know why the interface was blocked).

File: joblib/test/test_numpy_pickle.py [hunk 2]
```diff
@@ -425,23 +444,31 @@ def _check_pickle(filename, expected_list, mmap_mode=None):
...
+                        str(w.message)
+                        == "The file '{0}' has been generated with a joblib "
+                        "version less than 0.10. Please regenerate this "
+                        "pickle file.".format(filename)
+                    )
+                elif check_numpy_depreciation_warning and issubclass(
+                    w.category, np.exceptions.VisibleDeprecationWarning
+                ):
+                    assert (
+                        str(w.message)
```
[the remark this answers is not in the document]
Comment (Nanored4498) [reply joblib/test/test_numpy_pickle.py]: I will go for the simpler version then :)
